### ISSUE TITLE
Inconsistency in updating assets metadata in library

### ISSUE THREAD

**Issue by janagombitova** [akvo/akvo-lumen#1549]
# Context
With this issue: https://github.com/akvo/akvo-lumen/issues/778 we now add meta data to each asset in the library. We show user information (name or email depending on availability) and date and time. 

We have two ways how to assets meta data is now handled that causes an inconsistent approach and possible confusion. 

## Visualisations 
You edit an already existing visualisation and the changes are saved. The time updates to show the time of last edit and the user name changes to the user who did the last edit, not who created the visualisation. 

![user update - visualisation](https://user-images.githubusercontent.com/12456965/42620432-3e561b66-85bb-11e8-9365-73547aab25af.gif)

## Dashboards and datasets 
You edit an existing dashboard or dataset and the changes are saved. The time updates to show the time of last edit but the user name does not change and shows the creator, not the user who did the last edit. 

![user update - dataset](https://user-images.githubusercontent.com/12456965/42620440-416c6cd8-85bb-11e8-87e1-08b58f27c34c.gif)

# Problem or idea
This inconsistency in how user meta data is handled can confuse Jerome as: 
- for visualisation the user is handled as the user who made the latest change at the time and date showing for the visualisations 
- for dashboards and data sets it shows the creator, regardless of the edits that happened recently.

# Solution or next step
How could we solve it?

This issue brings up an overarching question: **With the meta data in the Library are we showing the user and time of creation or are we showing the user and time of last edit made to the entity?** If we go for one or the other, we should make sure the rule applies in the same way for all entities in the library on all the meta data.

For now, I would propose to take the approach of - we show the last edit meta data - so the user who made the change and the time of change, to ensure all is consistent before we decide how to handle the above question. 

@finnfiddle and @nadiagorchakova what do you think?

**Comment by nadiagorchakova** [comment]
In my opinion, for Jerome, it's more interesting to see the last edit made to the entity.
If he, as a project manager, has any questions about a vis/dashboard/dataset, he can reach out to the person who worked on it last and probably has most knowledge about it.
Perhaps it would make sense to show 'last edit: jerome@bla.org' in metadata, so that it's clear what we are showing. Also curious to know what Finn thinks.

**Comment by finnfiddle** [comment]
As I wrote to @janagombitova I looked at google drive and they have the `owner` and `last updated date` and then inside each entity they have a log of changes done by different users. I agree that its probably more useful to know who last edited something but I also think that if you are browsing through the library and you are looking for something in particular you might get confused if the person's name changes all the time. Im not sure tbh what is the best route. The change log would be a relatively big task I believe

**Comment by janagombitova** [comment]
Can we then for this iteration make sure the user reflects the `last updated` user also for data sets and visualisations to ensure the meta data is consistent and then in a separate issue discuss the possibilities of a change log also for visualisations and dashboards and if that would be of value?

**Comment by finnfiddle** [comment]
@tangrammer @kardan would it make sense to add a `lastUpdatedBy` field to each table similarly to the `author` field - or is it a better idea to add a separate table to store change logs for entities and just retrieve the latest each time?

**Comment by janagombitova** [comment]
With @finnfiddle and @tangrammer we agreed to revise the implementation of this issue to show author and time and date **of creation** rather than of last edit. 

We made this decision because we currently store the original author and the last modification date, not the last editor. To be able to push this change to users as quickly as we can we decided to go with the data we already store. 

Secondly, the entities in the library are organised by last edit date and time.  By showing this time stamp also in the meta data we duplicate information. 

We agreed to add to the UI `Create by` to make it clear what the meta data is about and get feedback to learn from users if this is helpful. @finnfiddle suggested a history log of edits for visualisations and dashboards to consider in the future.
